backend/clubs/views.py

```diff
@@ -4771,28 +4775,45 @@ def initiate_checkout(self, request, *args, **kwargs):
         )
 
         # Assert that the filter succeeded in freezing all the tickets for checkout
-        if tickets.count() != cart.tickets.all().count():
+        if tickets.count() != cart.tickets.all().count() or tickets.count() == 0:
             return Response(
                 {
                     "success": False,
-                    "detail": "Cart is stale, invoke /api/tickets/cart to refresh",
+                    "detail": (
+                        "Cart is stale or empty, invoke /api/tickets/cart to refresh"
+                    ),
+                    "sold_free_tickets": False,
                 },
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
         holding_expiration = timezone.now() + datetime.timedelta(minutes=10)
         tickets.update(holder=self.request.user, holding_expiration=holding_expiration)
 
-        cart_total = tickets.aggregate(total=Sum("price"))["total"]
-        if not cart_total:
+        # If all tickets are free, we can skip the payment process
+        if all(ticket.price == 0 for ticket in cart.tickets.all()):
```

Comment:
Thoughts on `if (cart_total := cart.tickets.aggregate(total=Sum("price"))["total"]) == 0:`? Since we're now checking if the ticket count is 0 beforehand, there shouldn't be a case where the cart total is 0 without the cart containing all free tickets

Separately, when we add `price` to `create_tickets`, we will need to add a check for non-negative ticket price

Reply:
I remember reading about the walrus operator, when it was released in 3.7/3.8 but never used it. Cool to use it here.

I see what you mean. Since `aggregate` returns `None` if there are no rows in the query result, `total` is mapped to `None`

Reply:
Good catch, added a check for non-negative ticket pricing in [065c60c](https://github.com/pennlabs/penn-clubs/pull/659/commits/065c60cfed3f966010ee2bd6d74ff04442ea745a).